Add Wiki for gsoc 2023 (#77)
oppia/oppia-web-developer-docs: Google-Summer-of-Code-2023.md

## Table of Contents

- [Getting started](#getting-started)
- [FAQs](#faqs)
- [Dates and Deadlines](#dates-and-deadlines)
- [Types of work related to Oppia projects](#types-of-work-related-to-oppia-projects)
- [GSoC proposal template](#gsoc-proposal-template)
  - [Tips for writing a good project plan](#tips-for-writing-a-good-project-plan)
  - [What should applicants expect from mentors in a proposal review?](#what-should-applicants-expect-from-mentors-in-a-proposal-review)
  - [Sample proposals from past years](#sample-proposals-from-past-years)
- [Selection Criteria](#selection-criteria)
- [Communication](#communication)
- [Oppia's Project Ideas List](#oppias-project-ideas-list)

> This year, Oppia will be participating in [Google Summer of Code 2023](https://summerofcode.withgoogle.com/)! GSoC is a global program which offers non-experienced contributors (and, historically, post-secondary students) the opportunity to discover and work with open-source organizations. The contributions are supported by a stipend. Contributors work closely with one or more mentors from an open-source organization to implement either a project idea by the organization, or a proposal of their own. You might be interested in our GSoC info pages from previous years: [2022](https://github.com/oppia/oppia/wiki/Google-Summer-of-Code-2022), [2021](https://github.com/oppia/oppia/wiki/Google-Summer-of-Code-2021), [2020](https://github.com/oppia/oppia/wiki/Google-Summer-of-Code-2020), [2019](https://github.com/oppia/oppia/wiki/Google-Summer-of-Code-2019), [2018](https://github.com/oppia/oppia/wiki/Google-Summer-of-Code-2018), [2017](https://github.com/oppia/oppia/wiki/Google-Summer-of-Code-2017), [2016](https://github.com/oppia/oppia/wiki/Google-Summer-of-Code-2016).

Reviewer: We can change the second sentence of this, since students are explicitly allowed in 2023. See https://developers.google.com/open-source/gsoc/faq -- "GSoC is for new and beginner contributors to open source and to students".

Maybe we can write: "which offers students and non-experienced contributors the opportunity..."

Author: done

In order to receive updates about GSoC at Oppia, please subscribe to the [Oppia GSoC Announce](https://groups.google.com/g/oppia-gsoc-announce) mailing list.

> Oppia will strictly adhere to GSoC timeline and will not be able to provide any additional extensions. Please refer to 

Reviewer: This may need more elaboration since we are thinking about giving an "automatic extension" and permitting an 8 + 8 week chunk (so it's not quite the "standard gsoc period"). The wording is also potentially ambiguous since there are now multiple "standard gsoc periods". So I think we will need to describe this quite carefully, probably in a separate section, so that there is no confusion about the timeline.

Alternatively, it might be better to put this information in the proposal template when applicants are proposing milestones. Let me know if you want to go this route and I'll update it there instead.

Author: Yeah, I have elaborated on this as well as in the dates section

-   [Google Summer of Code contributor guide](https://google.github.io/gsocguides/student/)
-   [Google's list of resources](https://developers.google.com/open-source/gsoc/resources/)
-   [GSoC FAQ](https://developers.google.com/open-source/gsoc/faq)

## Getting started

Welcome! If you're interested in applying to work with Oppia for GSoC, please follow these steps:

1. Sign up to the [oppia-gsoc-announce@](https://groups.google.com/forum/#!forum/oppia-gsoc-announce) mailing list in order to receive important notifications about Oppia's participation in GSoC. If you like, you can turn on the notification for github discussions to  participate in general discussion related to Oppia's involvement in GSoC. Make sure to set your preferences correctly so that you actually get the emails!

2. Get a better understanding of what Oppia is about:
    - Read the [user documentation](http://oppia.github.io/#/) to become familiar with important concepts like explorations and interactions.
    - Play some lessons on [Oppia.org](https://www.oppia.org/learn/math), which hosts a live instance of Oppia.

> 3. To get started with development, read and follow the instructions in the contributors' guide carefully ([Oppia Web](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#setting-things-up)).

Reviewer: This seems like a good place to call out Oppia Android's non-participation this year. 

Maybe change this last part to: "...in the Oppia Web contributors' guide carefully. (Note that Oppia Android won't be offering GSoC projects in 2023, but will be doing so again from 2024 onwards.)" with the link put on "Oppia Web contributors' guide".

Author: Added

4. Do one or more starter projects to become familiar with the contribution process. This will help us get an idea of what it's like to work with you. It will also help you get a better understanding of the codebase and our development process, which may help with writing a good project proposal. Once you've merged at least 2 pull requests, you will get an invitation to become a collaborator to the Oppia or Oppia-Android repository and be officially onboarded!

   - **Pro-tip!** Quality is more important than quantity; we want to see examples of your best work. So, please make sure to follow the [tips for success](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#tips-for-success), manually test your code before submitting (to ensure it does what you want it to and doesn't break anything else), ensure that your code conforms to the [style rules](https://github.com/oppia/oppia/wiki/Coding-style-guide), and pay attention to small details. These are good skills to learn when developing software in general, and they will also help you build credibility as a responsible developer who can be trusted to be a good steward of the Oppia codebase.

> 5. Select one or more [GSoC project ideas](#oppias-project-ideas-list) that you're most interested in, and write your project proposal! We strongly encourage you to discuss your project ideas and share your proposal with the community, so that you can get feedback and ensure that what you're writing makes sense to others. Please follow the instructions in the [GSoC proposal template](#gsoc-proposal-template) section to do this.

Reviewer 2: Should we specify that being onboarded is one of our requirements for GSoC?

Author: Added that to point 4

    If you have any questions about a project, or would like to discuss your approach with the Oppia community and get feedback, you can also email the [oppia-gsoc-discuss@](https://groups.google.com/g/oppia-gsoc-discuss) mailing list. Please be specific when asking questions, since this makes it easier for us to help you.
    - **Pro-tip!** 75% of the accepted proposals had begun their work at least 4 weeks prior to the deadline. Hence, we strongly recommend that the contributors start writing proposals as early as possible.

    Good luck!

## FAQs

**Q: What technical skills do I need to work on Oppia?**

> A: For Oppia Web, Angular 2+, Python 3.7 and Google App Engine is useful and recommended for most Oppia work; in addition, UI design skills are useful for frontend, user-facing work. For Oppia Android, you'll want to be familiar with Android development in Kotlin. Please see the individual project ideas to determine whether these skills are recommended for the project in question.

Reviewer 2: ```suggestion
A: For Oppia Web, Angular 2+, Python 3.8 and Google App Engine are useful and recommended for most Oppia work; in addition, UI design skills are useful for frontend, user-facing work. For Oppia Android, you'll want to be familiar with Android development in Kotlin. Please see the individual project ideas to determine whether these skills are recommended for the project in question.
```

Author: done, and removed Oppia Android from this answer

**Q: How can I increase my chances of getting selected?**

A: Writing a good project proposal, engaging with the community, helping other contributors, successfully contributing, and demonstrating that you can work independently can all help you. We've also compiled some notes below on the [selection criteria](#selection-criteria) we'll be using this year.

**Q: Can you be flexible around my other commitments in the summer?**

> A: Yes (within reason)! This year, GSoC is going to be a bit different from previous years. The program has been restructured in a way that allows contributors to take some time off for any other commitments such as exams. ([View the timeline here.](https://developers.google.com/open-source/gsoc/timeline)) Oppia will respect the same timelines that are given by GSoC; the main concern is whether you can still get the project done on time. Be upfront about your other commitments and make sure you schedule your time accordingly when creating your proposal. Other commitments you should list include time when you'll be in school and will commit less time to GSoC, time when you'll be travelling and away from GSoC work, any summer jobs you need to commit to, etc. We will try to be flexible around other time commitments, as long as your proposal convinces us that you will have enough time to complete the project by the end of the GSoC coding period. On the other hand, if you do not disclose other commitments, and it turns out that you are unable to commit to what you wrote on your proposal, this is grounds for failing the program.

Reviewer 2: We're not going to be so flexible this time around. Really, I think the answer to this question is "no" in that we aren't going to adapt the timeline based on the contributor's schedule based on the mess that created last year

Author: Updated it

**Q: Which projects are most important for Oppia?**

A: All the projects we've listed in the [Ideas List](#oppias-project-ideas-list) are important, and we'd be very happy to see good progress made on any of them! Projects are treated as equally important during selection; note that the relative importance of a project to Oppia is not part of the [selection criteria](#selection-criteria). We strongly encourage you to pick a project that you'd enjoy doing over the summer!

**Q: Is Oppia Android a part of this year's GSoC?**

> A: Oppia Android won't be able to participate in GSoC this year. Note that this is a one-off, and Oppia Android hopes to be back in 2024 when we have better processes and documentation to help contributors get onboarded more smoothly.

Reviewer 2: We should remove references to Android from the rest of this page (e.g. the technical skills FAQ)

Reviewer: We should also probably call this out more prominently or make it the top FAQ, since it is a significant change.

**Q: Can I submit more than one proposal to Oppia?**

> A: Yes, you can. However, we strongly recommend picking one project and writing a solid proposal for it. Splitting attention across multiple projects might not be a great idea. (That said, since this is the first year GSoC is offering full-length and half-length projects, one exception might be if you're interested in doing either the 'full version' or the 'half version' of a project idea that can support both modes. In such a case, you would be welcome to submit both the 'full version' and the 'half version' as separate applications, but, before doing so, please make sure that you'd be happy with either outcome if you are selected.)

Reviewer 2: This is no longer the first year of having two project lengths

Author: done

**Q: How early should I start working on the proposal?**

> A: As early as possible. Make sure to get feedback from mentors before finally submitting the proposal. This will help you to write a better proposal, as you can refine the details based on the feedback you receive. Mentors will need some time to review your proposal, so it's a good idea to begin as early as possible. Make sure to follow all instructions in the [proposal template](https://docs.google.com/document/d/1yYefLkT7dJJa86MyrdWpbZtzeaWAKCi1eXZZDGUrasM/edit) (especially around sharing and access) to reduce delays in reviewing your proposal.

Reviewer: This is a good place to reiterate the 75% stat mentioned in the improvements doc.

Author: added

**Q: I only discovered Oppia recently. Does this mean that, during selection, my application would automatically be ranked lower than those by other applicants who have a longer tenure with Oppia?**

A: Definitely not! Here are the [selection criteria](#selection-criteria) we use when selecting contributors for GSoC. Note that tenure is explicitly not part of these criteria.

**Q: What are the minimum number of PRs that one should have?**

> A: You should have at least 2 PRs merged, but they don't need to be large. Beyond that, remember that quality is more important than quantity. It is better to submit a non-trivial PR rather than a one-line wording change. Start with starter issues, then prioritize ones that touch the area(s) of the codebase which are related to your project.

Reviewer: Maybe we can be more specific in the last part and encourage them to do other issues offered by the teams that are offering the project -- and add links here to point them to the "prioritized issues" views in the LaCE, CD and Dev Workflow project boards.

Author: Added links to the prioritized issue views of each team boards

A: As many as we think will succeed, though the Google GSoC admins may impose limits based on how they decide to distribute contributor slots among the different open-source organizations.

**Q: I do not have any experience in skill XYZ. Is some certification required?**

A: Try to work on good first issues and take courses online. In the field of software development, it is common to develop experience and expertise as you take up and complete projects successfully. We do not require any formal certification of particular skills.

**Q: Is it okay if I only focus on the frontend or backend?**

A: This probably depends on the project(s) you wish to apply for. However, note that the ability to be effective in both the frontend and backend will open up more opportunities for you, since projects often touch multiple layers of the stack.

**Q: The [Google GSoC FAQ](https://developers.google.com/open-source/gsoc/faq#can_someone_already_participating_in_open_source_be_a_gsoc_contributor) mentions that the program is only for new contributors. I have already contributed to Oppia and I have write access. Can I still participate?**

> A: The GSoC program is open to new and beginner contributors to open source. Some of our contributors with write access are still beginner contributors, whereas some of our other contributors with write access will not qualify because they are experienced contributors. If you have only recently received write access, or have been contributing to Oppia for less than a year, you are probably still a beginner contributor. If the previous sentence does not apply to you, and you want to know which group you fall into, please contact **@vojtechjelinek** or **@seanlip** for a decision.

Reviewer: The eligibility requirements have changed this year to explicitly allow students, so this might need to be updated.

You might also want to change the contact points to the GSoC admins for this year.

Author: done

## Dates and Deadlines

Noteworthy dates for 2023 ([Full Timeline](https://developers.google.com/open-source/gsoc/timeline)):

- **Jan 23 - Feb 7**: Mentoring organizations apply
- **Feb 22**: Mentoring organizations are announced
- **Mar 20 - Apr 4**: GSoC contributor application period
- **May 4**: Accepted GSoC contributors are announced
- **May 4 - May 28**: Community bonding period
- **May 29 - Nov 6**: GSoC contributors enjoy the summer by contributing code to their projects

> - **Nov 13**: GSoC officially ends

Reviewer 2: We aren't allowing projects to run this late, right?

Reviewer: Maybe here we can do something like (replacing the durations with the exact dates):

- First 8 weeks: Standard coding period for Milestone 1
- Next 8 weeks: Standard coding period for Milestone 2

and add a note explaining that the coding period dates are strict and intended to balance offering flexibility to contributors with the need on the org admin side to have a streamlined process and a more time-bounded GSoC period.

(We can use the proposal template to explain the finer details of the external evals etc. but I think  the above might be the simplest way to communicate our plans for the timeline.)

Author: I have corrected the dates according to the timeline sheet shared by @U8NWXD

    -   How to manage my time well, how to achieve deadlines especially when I got evaluations from external evaluators.
    -   How to properly plan a project before implementing it.
    -   Technical design skills (and validation of technical ideas).
    -   How to give, respond to and understand reviews.
- Communication and personal development
    - Putting forward my thoughts more systematically and deeply so that everyone can understand me well.
    - My conversation skills have surely improved and I feel more confident while joining online meetings. Also, I am more mindful of the deadlines and enjoyed solving challenges faced in this project.
    - How to write a good proposal.
    - How to work with a large community like this which is spread over different time zones.
    - How to reach out to people, work with them, and solve each other's problems.
    - I have become a better developer, not only in term of technical skills but in thinking of actual application of the product build and the edge case scenarios the user might face. Another major learning has been to speak up and ask for help or anything required which I previously was not too confident of.

> - Why students want to stay

Reviewer: students --> contributors

Author: done

      - The kind of support the complete community provides is extra-ordinary.
    - Giving back
      - "Everybody wants to do some good work but lets be honest, it is maybe the lazyness or the time issue that you are not able to do the type of work you always wanted to do, what can be a good way to give back to the community than to contributing here.
      - It makes me very happy that I'm a part of an organization which provides free education and I think the education is the biggest blessing we can give to one to make them stand on their feet.
      - I would love to be part of this org by knowing that maybe not much but yes I'm trying to make an impact and my contribution in the educational field. I really want to do this because where I come from there is not much of education.
      - Also the main reason to stay connected is the purpose the community serves. Providing education to those who do not have access to it helps me give back to the society.
    - Growth / learning:
      - Working with Oppia has really helped me grow as a developer and I would really like to stick around to gain even more experience of real world software development.

>       - Yes, the kind of work that Oppia does is really inspiring and there are a lot of opportunities to improve your skills be it be technical skills or leadership skills and most of all the people at Oppia are really fun to work with :)

Reviewer: "Yes, the ..." --> "The ..."

Author: done

      - I like working in Oppia since it not only helps me improve my coding skills but also helps me grow as an individual.

## GSoC Proposal Template

> When submitting a proposal, please use the provided [GSoC proposal template](https://docs.google.com/document/d/1yYefLkT7dJJa86MyrdWpbZtzeaWAKCi1eXZZDGUrasM). We will only consider proposals submitted using this template.

Author: Since we don't have the template ready for this year, should I keep a placeholder here?

Reviewer 2: I'd leave out the link and include a placeholder stating that the template isn't ready yet. That way no one starts working based on the old template

Reviewer: Here is the updated proposal template for 2023: https://docs.google.com/document/d/1ZZEJXCFgqPgPDuImKAW6EB9SMOS8cICPI2uCsVLww8w/edit# . Please review and leave any comments as you see fit -- but I think you can include it here.

Author: done

-   If you're confused about something, try to identify the point of confusion and ask have specific discussions about it, rather than simply agreeing to whatever is proposed. Don't rely on an "appeal to authority" (e.g. "I am doing it this way because reviewer XXX said so") — the rational analysis and thought that underlie the decision are what's important, so make sure that you understand and clearly communicate the reasons behind the decisions you make.
-   Note that the process Oppia uses to select GSoC contributors typically includes multiple independent mentors, most of whom will not have looked at the earlier versions of your submitted proposal. Your reviewer may or may not be involved in the final selection process, and it is definitely **not** the case that you need to implement all your reviewer's suggestions/requests in order to be selected. We recommend considering your reviewer as a friendly advisor who is available to help you and provide guidance, rather than the main future evaluator of your proposal.

### Sample proposals from past years

> If you'd like to get a sense of what a proposal might contain, please see our [GSoC 2022 page](https://github.com/oppia/oppia/wiki/Google-Summer-of-Code-2022) for examples of proposals that we accepted in 2021. However, please note that the [GSoC Proposal Template](#gsoc-proposal-template) has been completely revamped for 2022, so please be sure to follow the 2022 template.

Reviewer: 2021 should be 2022.

Maybe change the second sentence to "slightly updated" rather than "completely revamped". Also in the second sentence change 2022 to 2023 in both places.

Author: done

To receive important announcements and updates about GSoC at Oppia, please subscribe to the **[Oppia GSoC Announce](https://groups.google.com/g/oppia-gsoc-announce)** mailing list.

## Oppia's Project Ideas List

_**Note:** If you're coming to this section from an external link, please make sure to scroll up and read this entire wiki page carefully, not just this section. There's a lot of useful information on the rest of the page, including a FAQ and a section describing selection criteria. Thanks!_

The following is a list of Oppia's 2023 GSoC project ideas. You are welcome to choose among these ideas, or propose your own! However, if you're planning to propose something original, it's essential to engage with the Oppia community beforehand in order to get feedback and guidance to improve the proposal. We'd also recommend taking a look at [Oppia's mission](https://github.com/oppia/oppia/wiki/Oppia's-Mission) and seeing if there is a natural way to tie your idea to the Oppia project's goals, otherwise it might not be a good fit at this time.

> Please note that the list of project ideas below is not set in stone: more projects may be added later, and some project descriptions may also change a bit, so check back regularly. In addition, the mentor assignments listed below are provisional, and may change depending on which proposals are eventually accepted.

Reviewer 2: Maybe add a note here that projects are coming later, that way readers don't think we forgot to include them

Author: Added

In order to receive updates about GSoC at Oppia, please subscribe to the [Oppia GSoC Announce](https://groups.google.com/g/oppia-gsoc-announce) mailing list.

> Oppia will be giving an automatic extension to all the projects i.e, all the projects will have 8 weeks (6 weeks for coding and 2 weeks for revision after evaluation) for both milestones. The coding period dates are strict and intended to balance offering flexibility to contributors with the need on the org admin side to have a streamlined process and a more time-bounded GSoC period.

Reviewer: I think I would just say "...all the projects will have 8 weeks for each milestone, with an internal evaluation after 6 weeks" and drop the parenthetical. We can explain how it's divided later on but I wouldn't want to give the impression that "2 weeks for revision after evaluation" is standard -- most students ought to be done in 6 weeks and that should be the baseline expectation.

Additionally, maybe add "Contributors may opt for a shorter timeline if desired." at the end -- we don't want to cause a delay in stipends for contributors if they finish all their stuff well and on time.

Author: done. changed it

    Good luck!

## FAQs

**Q: Is Oppia Android a part of this year's GSoC?**

A: Oppia Android won't be able to participate in GSoC this year. Note that this is a one-off, and Oppia Android hopes to be back in 2024 when we have better processes and documentation to help contributors get onboarded more smoothly.

**Q: What technical skills do I need to work on Oppia?**

A: For Oppia Web, Angular 2+, Python 3.8 and Google App Engine are useful and recommended for most Oppia work; in addition, UI design skills are useful for frontend, user-facing work. Please see the individual project ideas to determine whether these skills are recommended for the project in question.

**Q: How can I increase my chances of getting selected?**

A: Writing a good project proposal, engaging with the community, helping other contributors, successfully contributing, and demonstrating that you can work independently can all help you. We've also compiled some notes below on the [selection criteria](#selection-criteria) we'll be using this year.

**Q: Can you be flexible around my other commitments in the summer?**

> A: No, Oppia will strictly adhere to GSoC timeline and will not be able to provide any additional extensions. Please refer to 

Reviewer 2: ```suggestion
A: No, Oppia will strictly adhere to GSoC timeline and will usually not be able to provide any additional extensions. Please refer to 
```

Just to give us a little wiggle room in case something extreme happens. Also, there shouldn't be a line break after "refer to"

Author: done. The line break was added to break the sentence. Visually when the wiki is added I don't think it will come over there 
<img width="514" alt="Screen Shot 2023-01-13 at 5 16 31 PM" src="https://user-images.githubusercontent.com/38973423/212429438-4df06834-a370-4f48-9451-b0a318c75937.png">

Reviewer 2: Yes, the line break doesn't affect the rendered text. However, our style in the wiki is to not break lines in the middle of a paragraph. This avoids the creation of large diffs from re-wrapping blocks of text when just a few words are changed.

Author: Oh okay, will do that

Author: done

the [Dates and Deadlines](#dates-and-deadlines) section below.

**Q: Which projects are most important for Oppia?**

A: All the projects we've listed in the [Ideas List](#oppias-project-ideas-list) are important, and we'd be very happy to see good progress made on any of them! Projects are treated as equally important during selection; note that the relative importance of a project to Oppia is not part of the [selection criteria](#selection-criteria). We strongly encourage you to pick a project that you'd enjoy doing over the summer!

**Q: Can I submit more than one proposal to Oppia?**

> A: Yes, you can. However, we strongly recommend picking one project and writing a solid proposal for it. Splitting attention across multiple projects might not be a great idea. (That said, GSoC is offering full-length and half-length projects, one exception might be if you're interested in doing either the 'full version' or the 'half version' of a project idea that can support both modes. In such a case, you would be welcome to submit both the 'full version' and the 'half version' as separate applications, but, before doing so, please make sure that you'd be happy with either outcome if you are selected.)

Reviewer 2: ```suggestion
A: Yes, you can. However, we strongly recommend picking one project and writing a solid proposal for it. Splitting attention across multiple projects might not be a great idea. (That said, GSoC is offering full-length and half-length projects, so one exception might be if you're interested in doing either the 'full version' or the 'half version' of a project idea that can support both modes. In such a case, you would be welcome to submit both the 'full version' and the 'half version' as separate applications, but, before doing so, please make sure that you'd be happy with either outcome if you are selected.)
```

Author: done

A: As many as we think will succeed, though the Google GSoC admins may impose limits based on how they decide to distribute contributor slots among the different open-source organizations.

**Q: I do not have any experience in skill XYZ. Is some certification required?**

A: Try to work on good first issues and take courses online. In the field of software development, it is common to develop experience and expertise as you take up and complete projects successfully. We do not require any formal certification of particular skills.

**Q: Is it okay if I only focus on the frontend or backend?**

A: This probably depends on the project(s) you wish to apply for. However, note that the ability to be effective in both the frontend and backend will open up more opportunities for you, since projects often touch multiple layers of the stack.

**Q: The [Google GSoC FAQ](https://developers.google.com/open-source/gsoc/faq#can_someone_already_participating_in_open_source_be_a_gsoc_contributor) mentions that the program is only for new contributors. I have already contributed to Oppia and I have write access. Can I still participate?**

> A: The GSoC program is open to new and beginner contributors to open source and to students. Some of our contributors with write access are still beginner contributors, whereas some of our other contributors with write access will not qualify because they are experienced contributors. If you have only recently received write access, or have been contributing to Oppia for less than a year, you are probably still a beginner contributor. If the previous sentence does not apply to you, and you want to know which group you fall into, please contact **@srijanreddy98**, @U8NWXD or **@aasiffaizal** for a decision.

Reviewer: Should you make U8NWXD's handle bold as well, like the other two?

Author: Oops, I missed that. I have made it bold

## Dates and Deadlines

Noteworthy dates for 2023 ([Full Timeline](https://developers.google.com/open-source/gsoc/timeline)):

- **Jan 23 - Feb 7**: Mentoring organizations apply
- **Feb 22**: Mentoring organizations are announced
- **Mar 20 - Apr 4**: GSoC contributor application period
- **May 4**: Accepted GSoC contributors are announced
- **May 4 - May 28**: Community bonding period
- **May 29 - Jul 14**: Milestone 1 work period for GSoC contributors

> - **Jul 15 - Jul 16**: External evaluation for M1

Reviewer: I wonder if we should call this the "internal evaluation" since the GSoC one is kind of the external one?

Author: Well, yeah. Internal evaluation makes more sense

> - **Jul 17 - Jul 26**: GSoC contributors can revise M1 according based on th evaluation

Reviewer 2: ```suggestion
- **Jul 17 - Jul 26**: GSoC contributors revise M1 according based on the evaluation
```

I'd remove the "can" since it's not really optional. Ditto below

Reviewer 2: nvm. already addressed

> - **Jul 17 - Jul 26**: GSoC contributors can revise M1 according based on th evaluation

Reviewer: th --> the ?

Also maybe we should write here: "Jul 17 - Jul 26: Optional buffer period for M1 revisions. (Contributors who have passed M1 can begin M2 during this phase.)" Otherwise this seems like a mandatory waiting time.

Author: Yeah changed to "Optional buffer period for M1 revisions"

ditto for M2

- **Jul 31 - Sep 15**: Milestone 2 work period for GSoC contributors
- **Sep 16 - Sep 17**: External evaluation for M2

> - **Sep 18 - Sep 27**: GSoC contributors can revise M2 according based on th evaluation

Reviewer: "Sep 18 - Sep 27: Optional buffer period for M2 revisions."

Author: done

> - **Oct 6**: GSoC officially ends

Reviewer: GSoC period at Oppia officially ends 

(since this doesn't describe GSoC as a whole)

Author: done

A: Yes, GSoC might not be the best choice if you don't have enough time during the summer. However, you can start contributing to Oppia by following the instruction in the contributors' guide carefully ([Oppia Web](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#setting-things-up), [Oppia Android](https://github.com/oppia/oppia-android/wiki/Contributing-to-Oppia-android)).

**Q: Why doesn't Oppia provide difficulty levels for projects?**

A: Difficulty can vary from person to person depending on the skills required which is why we list all the prerequisite skills of each project.
 
## Dates and Deadlines

Noteworthy dates for 2023 ([Full Timeline](https://developers.google.com/open-source/gsoc/timeline)):

- **Jan 23 - Feb 7**: Mentoring organizations apply
- **Feb 22**: Mentoring organizations are announced
- **Mar 20 - Apr 4**: GSoC contributor application period
- **May 4**: Accepted GSoC contributors are announced
- **May 4 - May 28**: Community bonding period
- **May 29 - Jul 14**: Milestone 1 work period for GSoC contributors
- **Jul 15 - Jul 16**: Internal evaluation for M1

> - **Jul 17 - Jul 26**: Optional buffer period for M1 revisions

Reviewer 2: "Optional" makes it sound like contributors can leave this buffer out of their proposal timelines, but I don't think we're allowing that. I think we can remove "optional" and just call it a "buffer period"

Reviewer: Actually, in their timelines, they shouldn't be planning anything for the "buffer period", so the proposal timelines shouldn't include it, right?

Reviewer 2: Yeah, maybe we explicitly say that they should only be planning revisions for that period

Reviewer: Hm. I suggest that we keep this part high-level, and put more details in the proposal template, which would be easier to edit if needed. (I've already put some details there, actually.)

For this PR, do you still dislike the current text? Not quite sure how to parse your previous comment with respect to that.

Reviewer 2: I do still disagree with the current text. "Optional" makes it sound like students can opt to skip the buffer period and, for example, plan to start M2 on Jul 17. That's misleading. Instead, the buffer period is a required part of our project timeline for this year.

I can see why one might want to include the word "optional" to convey that if no issues are raised during the evaluation, the student can move ahead. However, this doesn't really make the buffer period "optional" for the student since its dependent on a decision by the evaluator. Further, this property of the student being able to work ahead of schedule is true throughout the project, so it's inconsistent to only call it out for the buffer periods.

I think we should drop the word "optional" since the buffer period is a required part of our timeline. I'm fine with leaving this timeline high-level and not delving into what students do during the buffer period, but I don't think we should include "optional" since that implies something incorrect

Reviewer: One other suggestion (y'all might want to discuss): you could just collapse these three lines and say:

- "May 29 - Jul 26": Milestone 1 work period (or "standard Milestone 1 work period")
- "... - ..." : ditto (for M2)

and leave all the other stuff out of it, since it can be (and has been) elaborated on in the proposal template.

Author: I think let's go with the collapsed timeline. I have collapsed the three lines to "Standard Milestone X work period for GSoC contributors"

**Q: The [Google GSoC FAQ](https://developers.google.com/open-source/gsoc/faq#can_someone_already_participating_in_open_source_be_a_gsoc_contributor) mentions that the program is only for new contributors. I have already contributed to Oppia and I have write access. Can I still participate?**

A: The GSoC program is open to new and beginner contributors to open source and to students. Some of our contributors with write access are still beginner contributors, whereas some of our other contributors with write access will not qualify because they are experienced contributors. If you have only recently received write access, or have been contributing to Oppia for less than a year, you are probably still a beginner contributor. If the previous sentence does not apply to you, and you want to know which group you fall into, please contact **@srijanreddy98**, **@U8NWXD** or **@aasiffaizal** for a decision.

**Q: I'd love to contribute to open source, but I'm not sure I have enough time during the summer to do a GSoC project. Can I still help out?**

> A: Yes, GSoC might not be the best choice if you don't have enough time during the summer. However, you can start contributing to Oppia by following the instruction in the contributors' guide carefully ([Oppia Web](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#setting-things-up), [Oppia Android](https://github.com/oppia/oppia-android/wiki/Contributing-to-Oppia-android)).

Reviewer: instruction --> instructions

Probably ok to strike "carefully" here.

Author: done

**Q: Why doesn't Oppia provide difficulty levels for projects?**

> A: Difficulty can vary from person to person depending on the skills required which is why we list all the prerequisite skills of each project.

Reviewer: ...skills required. This is why we instead list the prerequisite ...

of --> for

Author: done
